src/transformers/trainer.py

```diff
@@ -2155,7 +2155,7 @@ def _inner_training_loop(
             self.state = TrainerState.load_from_json(os.path.join(resume_from_checkpoint, TRAINER_STATE_NAME))
             self.compare_trainer_and_checkpoint_args(self.args, self.state)
             self._load_callback_state()
-            epochs_trained = self.state.global_step // num_update_steps_per_epoch
+            epochs_trained = int(self.state.global_step // num_update_steps_per_epoch)
```

Comment:
Nice catch, if `global_step` or `num_update_steps_per_epoch` is a `float`, `//` doesn't cast to int. Thanks!